controllers/actions.github.com/autoscalinglistener_controller.go

```diff
@@ -86,7 +86,7 @@ func (r *AutoscalingListenerReconciler) Reconcile(ctx context.Context, req ctrl.
 		}
 		if !done {
 			log.Info("Waiting for resources to be deleted before removing finalizer")
-			return ctrl.Result{}, nil
+			return ctrl.Result{Requeue: true}, nil
```

Comment:
we need to requeue ourselves for every place that we rely on the Secret is updated/created/deleted

Reply:
Sounds really easy to forget 😬 do we have an easy way to ensure that doesn't happen?

Reply:
As far as I know, we can't :disappointed: 
This is artificially enqueueing the same event, so we get triggered again
Since we are not watching on those resources, we would have to trigger ourselves each time we do some operation...

Reply:
Our controller L0 test will catch it when you forgot requeue, I was able to use L0 to help me found that you added a new secret for proxy in listener and I need to requeue on that secret creation.